### DISCUSSION TITLE
Long loading times, up to 10 seconds before anything loads on the page.

### DISCUSSION THREAD

**Discussion by stellarfinn** [Budibase/budibase#10595] [Help] [answered]
Hosted on: Budibase Cloud
Design: Simple, single New Table Block

See below screenshots. For the life of me I don't understand why the it's taking 6.6 seconds to load budibase-client.js, and then another 10+ seconds to fetch tother things initiated by budibase-client.js.

Screenshots attached below. 
Currently paying for Budibase Cloud hosting and my boss is getting on me about load times. Any help very much appreciated. 

![Pricing Waterfall](https://github.com/Budibase/budibase/assets/99847728/6b046a14-48b9-446c-8483-897d1e3fe8fd)
![Pricing Design](https://github.com/Budibase/budibase/assets/99847728/b28cc39f-2a7f-484d-be15-9a04d53c1e6f)
![Pricing Table Block Conditions](https://github.com/Budibase/budibase/assets/99847728/7e4eec41-f0a6-4257-b069-f8477f37ad43)

**Comment by shogunpurple** [comment]
Have you throttled your own network connection?

![Screenshot 2023-05-15 at 21 29 42](https://github.com/Budibase/budibase/assets/11256663/e4859155-99dc-4372-9516-78b174bd5ec2)

Set this
![Screenshot 2023-05-15 at 21 30 10](https://github.com/Budibase/budibase/assets/11256663/bfea4e56-2f5e-4654-bd3e-39bd7241738f)

**Reply by stellarfinn** [reply]
Did what you suggested, results of Waterfall below....

<img width="880" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/ba502490-beff-47e9-99f3-ab3ddec3048a">

**Reply by shogunpurple** [reply]
It appears the budibase-client and other static file issues are looking much better - the setting you had on manually throttles your network connection to make it slower, which is just a testing tool. 

I'm having trouble recreating this - `/self` loading quickly for me in all my envs, and from checking our monitoring tools it appears the slowest call made to `/self` across our entire cloud in the past hour was 236ms. 

Can you tell me a bit more about your network environment? Could you have any add ons/VPN type connections that might be slowing this down?

**Reply by stellarfinn** [reply]
shogun, 

Not that I'm aware of. I don't use a VPN. Only other thing I can think of is Google Drive.

**Reply by stellarfinn** [reply]
Also, it's unique to Budibase... there's not single other website that loads slowly. I'm in Colorado, the office is South Florida and we're all experiencing long load times. Long meaning 10+ seconds throughout our Budibase application.

**Reply by shogunpurple** [reply]
@stellarfinn when did this start, just today?

**Reply by stellarfinn** [reply]
There's been a pretty consistent slow load time for a while. Even worse when they moved away from the spinning loader. It's improved since reverting back to the spinning loader but generally any page that has database calls takes a minimum of 5-10 seconds to load and sometimes much longer. 

Here's another example:
<img width="1915" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/1db4634f-9577-4afc-892e-334e31de63d5">

**Reply by shogunpurple** [reply]
Would it be possible to get an export of your app sent to support@budibase.com? It would really help us debug in this situation.

**Reply by stellarfinn** [reply]
Just emailed the published export. Thanks for your help!

**Reply by shogunpurple** [reply]
Is it a certain page in particular? Seeing better loading times on my end. 

![Screenshot 2023-05-15 at 22 07 27](https://github.com/Budibase/budibase/assets/11256663/2c643bb1-32de-4313-81c4-2b8927d6c69b)

There's a possibility this could be latency related given you are in the US, but if it used to work as expected that would rule this out.

**Reply by stellarfinn** [reply]
Kind of the whole thing in general, occasionally some pages are quick, but the more complicated the page the load times get to the absurd 20+ second timeframe. Even this front page which is pretty simple came in long.

Just tried the same page twice, first came in at 18.27s Finish, second at 13.66s Finish. 

<img width="1916" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/7add7b83-7c1b-45a9-8ff8-541c0e8a777d">

<img width="1912" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/4d21d241-5722-4cdd-a7e7-6d7a69bf2af4">

**Reply by stellarfinn** [reply]
I've made the changes you suggested, seriously simplified it down and it still runs very slow. 11 seconds to load my Frontpage.

https://ciaoimports.budibase.app/app/portal

I'm at a loss. The simplest pages take too long to load.

**Comment by stellarfinn** [comment]
Here's a screen recording to show how poorly the experience is... having a hard time justifying this any longer to my boss. 

https://github.com/Budibase/budibase/assets/99847728/635e51da-82e1-45b8-b211-b54247d96815

**Comment by stellarfinn** [comment]
@shogunpurple @mjashanks 

Here's a video of the homepage loading... and a screenshot of the builder view. Any help would be really appreciated I've been pushing Budibase as THE solution for a year and these load times have just gotten worse and worse. Is Budibase cloud servers not mirrored in the US? Is that the problem? Latency?

https://github.com/Budibase/budibase/assets/99847728/19d67cf1-a5d9-4ad8-9725-aefca53d6921

<img width="1907" alt="image" src="https://github.com/Budibase/budibase/assets/99847728/c4cd021f-0c14-4278-bec9-7108a347de53">

**Comment by stellarfinn** [comment, marked as answer]
SOLVED. 

The issue is formula fields that are dynamic. Each additional formula field that functions dynamically increases the load time. I removed almost all but a couple and the load time is now a reasonable period. No idea why you guys were able to see it loading faster than me but this has solved the issue for me. Took duplicating the app, deleting tables one at a time to identify the table causing the problem, then reviewing each formula field for dynamic setting. Once I realized this was the issue, I returned to my original app and began working on setting all formulas that were dynamic, to static. Then working through issues where not having dynamic formulas messes things up.
